.ladle/components.tsx

```diff
@@ -1,17 +1,44 @@
 import * as React from 'react';
-import {Provider as StyletronProvider} from 'styletron-react';
-import {Client as Styletron} from 'styletron-engine-atomic';
-import {LightTheme, DarkTheme} from '../src/themes/index.js';
+import { Provider as StyletronProvider } from 'styletron-react';
+import { Client as Styletron } from 'styletron-engine-atomic';
+import { LightTheme, DarkTheme } from '../src/themes/index.js';
 import BaseProvider from '../src/helpers/base-provider.js';
 
-const engine = new Styletron();
+let prevWidth = 0;
+
+// finds the closest document.head (one in iframe or top-level)
+const getDocument = (story: string) => {
+  const iframe = document.querySelector(`[title='Story ${story}']`) as HTMLIFrameElement;
+  return iframe && iframe.contentDocument ? iframe.contentDocument : document;
+};
 
 export const Provider: React.FC<{
   globalState: {
     theme: string;
     rtl: string;
+    story: string;
+    width: number;
   };
-}> = ({children, globalState}) => {
+}> = ({ children, globalState }) => {
+  const [engine, setEngine] = React.useState(
+    new Styletron({ container: getDocument(globalState.story).head })
```

Comment:
```suggestion
    () => new Styletron({ container: getDocument(globalState.story).head })
```

Does constructing have any side effects? https://reactjs.org/docs/hooks-reference.html